cmd/tools/check/blocks.go

```diff
@@ -231,14 +232,20 @@ func validateBlockSegment[B firecore.Block](
 			seenBlockCount++
 
 			if printDetails == PrintStats {
-				err := print2.PrintBStreamBlock(block, false, os.Stdout)
+				err := printer.PrintTo(block, os.Stdout)
 				if err != nil {
 					fmt.Printf("❌ Unable to print block %s: %s\n", block.AsRef(), err)
 					continue
 				}
 			}
 
 			if printDetails == PrintFull {
+				printer, err := print2.GetOutputPrinter(globalToolsCheckCmd, chain.BlockFileDescriptor())
```

Comment:
I plan to remove all printing support from `check`.

Reply:
agreed.